### PR TITLE
Define a simpler PenguinDataset in Exercise 1 notebook

### LINKED ISSUE #71: Simpler PenguinDataset
To make the data reading aspect a little easier to understand, we intend to embed a simpler version of the `PenguinDataset` directly into the notebook. The `src/ml_workshop/_penguins.py` will remain untouched and can still be used as before. 

Thought process around the 'simpler' class (discussion between @jatkinson1000 and @ma595)

Load pandas df in notebook:

Put the definition of the PenguinDataset in the notebook:
- Remove x_tfms and y_tfms
- Hardcode one_hot, tensor, fp32 into `__getitem__`

Propagate change to
- solution notebook
- colab
- ex 2+? Consider leaving this as is.

### PR DESCRIPTION
**Pull request by ma595**
This PR creates a simpler `PenguinDataset` within the `worked-solutions/01_penguin_classification_solutions.ipynb` notebook.

Within this easier PenguinDataset we:

- Move ontents of `_load_penguin_data(...)` and `_split_data(...)` into `__init__()`
- Remove `lambdas` and `Compose` object, making it easier to follow.
- Remove docstrings (I am aware that this isn't ideal, but I think it is fine for this purpose). Any objections, please comment.

Other notebook changes include:
- The unchanged `_penguins.py` has been referenced in the notebook as an example of good software practice.
- Markdown has been modified to make the notebook easier to follow in light of the above changes.
  - Mention of why data reading code is encapsulated in a class  
- All discussion on `Compose` has been removed. 

Before merging, the following tasks must be completed:
Propagate changes to:
- [x] exercise notebook
- [x] colab - (is **NOT** done automatically - This is done by rebasing or merging into the colab branch - consider doing this after merging to main.)
- [ ] ex 2+? Consider leaving this as is.

Closes #71

### CODE REVIEW

**Comment by ma595** [comment]
@jatkinson1000 would you prefer the jupyter notebook outputs to be retained?

**Comment by ma595** [comment]
Jack confirmed that we should retain the cell outputs of the solutions notebook for reference.

**Review by jatkinson1000 — COMMENTED** [review]
This all looks sensible to me in theory @ma595 

Caveat that I have not run the code and will leave this (and approval) to @surbhigoel77 as the co-presenter.

I agree that it would be easier to move changes to the colab branch after merging to main.

Also consider how other exercises might need adapting.

**Review by surbhigoel77 — APPROVED** [review]
Runs fine for me,

File: exercises/01_penguin_classification.ipynb [hunk 1]
```diff
@@ -126,14 +242,46 @@
...
+    "  - During training, we monitor the model's performance on the validation set in order to check how it's doing on unseen data. Normally, people use the validation performance to determine when to stop the training process.\n",
+    "  \n",
+    "For the validation set, we choose ten males and ten females of each species. This means the validation set is less likely to be biased by sex and species, and is potentially a more reliable measure of performance. You should always be _very_ careful when choosing metrics and splitting data.\n",
+    "\n",
+    "- Is this solution general?\n",
     "\n",
-    "These transforms can be very useful for mapping between file paths and tensors of images, etc.\n",
+    "A common way of transforming inputs to neural networks is to apply a series of transforms using `torchvision.transforms.Compose`. The [ `Compose` ](https://pytorch.org/vision/stable/generated/torchvision.transforms.Compose.html) object takes a list of callable objects and applies them to the incoming data. See how this is done more generally in the `src/ml_workshop/_penguins.py` file. \n",
```
[the remark this answers is not in the document]
Comment (ma595) [reply exercises/01_penguin_classification.ipynb:259]: Yes, it is.

**Pull request merged**